templates/skeleton/app/components/Header.tsx

```diff
@@ -159,15 +163,19 @@ function CartToggle({cart}: Pick<HeaderProps, 'cart'>) {
   return (
     <Suspense fallback={<CartBadge count={null} />}>
       <Await resolve={cart}>
-        {(cart) => {
-          if (!cart) return <CartBadge count={0} />;
-          return <CartBadge count={cart.totalQuantity || 0} />;
-        }}
+        <CartBanner />
       </Await>
     </Suspense>
   );
 }
 
+function CartBanner() {
+  const originalCart = useAsyncValue() as CartApiQueryFragment | null;
+  const cart = useOptimisticCart(originalCart);
+  const linesCount = cart.totalQuantity ?? null;
```

Comment:
Cart would be `null` here on first load.

And nit: `totalQuality` would be more consistent (vs `linesCount`).